src/lib.rs

```diff
@@ -192,59 +312,136 @@ impl<'object, Endian> DebugInfo<'object, Endian>
                 }
             }
         }
-        None
+
+        loc.map(move |(f, l)| (f, l, func.map(|u| u.0.name.to_string_lossy())))
     }
 }
 
-// TODO: most of this should be moved to the main library.
-struct Unit<'input> {
-    address_size: u8,
-    ranges: Vec<gimli::Range>,
-    line_offset: gimli::DebugLineOffset,
-    comp_dir: Option<&'input std::ffi::CStr>,
-    comp_name: Option<&'input std::ffi::CStr>,
+enum Target<'input> {
+    Line(LineUnit<'input>),
+    Function(FunctionUnit<'input>),
```

Comment:
I guess they could be separate, though this provides a neat way to share functionality between them (like looking up ranges). I guess we could provide something similar with a wrapper type though. Is there a particular reason you'd like to see them separated?